### PR TITLE
issue #2284 - support skippable updates for batch/transaction methods

### PR DESCRIPTION
**Pull request by lmsurpre**
If `X-FHIR-UPDATE-IF-MODIFIED` is set to true on the request, then all
updates (including patches) in the bundle will be processed in accordance 
with this update optimization.

Signed-off-by: Lee Surprenant <lmsurpre@us.ibm.com>

### CODE REVIEW

File: fhir-path/src/main/java/com/ibm/fhir/path/util/FHIRPathUtil.java [hunk 1]
```diff
@@ -905,6 +905,11 @@ public static String unescape(String s) {
 
         FHIRPathTree tree = evaluator.getEvaluationContext().getTree();
         FHIRPathNode parentNode = tree.getParent(node);
+
+        if (parentNode == null) {
+            throw new FHIRPatchException("Unable to compute the parent for '" + elementName + "';" +
+                    " a FHIRPathPatch replace FHIRPath must select a node with a parent", fhirPath);
+        }
```
Comment (lmsurpre) [inline fhir-path/src/main/java/com/ibm/fhir/path/util/FHIRPathUtil.java:912]: this is a somewhat unrelated change that might merit its own separate issue, but i found this NPE bug when I added a patch test and got the patch wrong

File: fhir-server-test/src/test/java/com/ibm/fhir/server/test/BundleTest.java [hunk 2]
```diff
@@ -2331,6 +2334,94 @@ public void testTransactionConditionalDeletes() throws Exception {
         assertGoodGetResponse(responseBundle.getEntry().get(1), Status.OK.getStatusCode(), HTTPReturnPreference.MINIMAL);
     }
 
+    /**
+     * Test transaction bundle put with post local reference dependency.
```
Comment (michaelwschroeder) [inline fhir-server-test/src/test/java/com/ibm/fhir/server/test/BundleTest.java:2338]: Update testcase description to match what is being tested.

**Review by michaelwschroeder — APPROVED** [review]
LGTM

File: fhir-server-test/src/test/java/com/ibm/fhir/server/test/BundleTest.java (hunk 2 above)
Comment (lmsurpre) [reply fhir-server-test/src/test/java/com/ibm/fhir/server/test/BundleTest.java:2338]:
```suggestion
     * Sets UPDATE_IF_MODIFIED_HEADER_NAME  and posts a transaction bundle with an update and a patch; both should be skipped on the server
```

**Review by JohnTimm — APPROVED** [review]
LGTM
